elliptic-curve/src/public_key.rs

```diff
@@ -177,8 +179,10 @@ where
     UntaggedPointSize<C>: Add<U1> + ArrayLength<u8>,
     UncompressedPointSize<C>: ArrayLength<u8>,
 {
-    fn from(public_key: &PublicKey<C>) -> EncodedPoint<C> {
-        public_key.to_encoded_point(C::COMPRESS_POINTS)
+    type Error = Error;
+
+    fn try_from(public_key: &PublicKey<C>) -> Result<EncodedPoint<C>, Error> {
+        public_key.to_encoded_point(C::COMPRESS_POINTS).ok_or(Error)
     }
```

Comment:
These conversions could potentially remain infallible if we had `PublicKey` enforce the invariant that the inner point cannot be the additive identity.

That would require making `PublicKey::from_affine` fallible, which might make sense, and would nicely match the similar invariants of `SecretKey`, which wraps a `NonZeroScalar`.

All that said, I'd like to follow up on that in a separate PR.